### ISSUE TITLE
Instantiating with new self() complains about missing parenthesis while the are there

### ISSUE THREAD

**Issue by stof** [squizlabs/PHP_CodeSniffer#1365]
I just update PHP_CodeSniffer from 2.7.1 to 2.8.0 in our project, and I got a few weird failures.

```
----------------------------------------------------------------------

FOUND 1 ERROR AFFECTING 1 LINE

----------------------------------------------------------------------

 63 | ERROR | Use parentheses when instantiating classes

----------------------------------------------------------------------
```

The line 63 looks like this:

```php
    public static function fromRankingSubjectAndScore(Subject $subject, $score, $rank)
    {
        return new self($subject, $score, null, null, $rank); //  <-- this is line 63
    }
```

This was working fine in 2.7.1

**Comment by gsherwood** [comment]
This doesn't look like a built-in sniff. If you add `-s` to the command, the report will show you what sniff is generating the error. If it is a custom sniff, you might need to speak to the author of that sniff to find out what is going on.

**Comment by stof** [comment]
Indeed. The issue comes from https://github.com/djoos/Symfony2-coding-standard/blob/master/Symfony2/Sniffs/Objects/ObjectInstantiationSniff.php

It is broken because it does not account for the bugfix of https://github.com/squizlabs/PHP_CodeSniffer/issues/1245

**Comment by stof** [comment]
thanks for the `-s` option. I was not aware of it.
